### ISSUE TITLE
Inject accounts into `web3.eth.accounts` and web3.eth.getAccounts()`.

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/web3.py#853]
### What was wrong?

It's useful to use `web3.eth.accounts` for things like:

- list of accounts to select from.
- pre-check for some pre-condition that is account based.
- auto selection for some automated criteria.

Acocunts used with the new in-flight transaction signing middleware will not show up in this list.

### How can it be fixed?

Add a conditional to the constructor for the transaction signing middleware that when set truthy will inject any addresses it can sign for into the returned accounts list.

link #649

**Comment by owocki** [comment]
@pipermerriam is this something team gitcoin should/could bounty?

**Comment by pipermerriam** [comment]
@owocki yes, but it's dependent on #649 being merged as well as a bit of confirmation from @dylanjw and/or @carver on whether they agree with how this should be implemented.

**Comment by carver** [comment]
Yeah, that sounds good to me. I'd even advocate to drop the flag altogether, and always inject the addresses, until someone requests it. It's much easier to add the flag in a minor point release than to remove it at the next major release, if we decide we don't want to support it.

Also, if `strict` is on, maybe we should strip out the node's addresses, since we can't use them anyway.

**Comment by pipermerriam** [comment]
I think I'm +1 on address injection, though I am inclined to advocate for the same solution I suggest in #649 which is to implement the address injection as a stand-alone middleware and to write a new utility function `merge_middlewares` or `combine_middlewares` which wraps up multiple middlewares into a single unit.

note that `combine_middlewares` is already a function we use in` web3.middlewares.combine_middlwares`, though I'd advocate for renaming that function to something like `compile_middlewares`

**Comment by dylanjw** [comment]
I agree with the middleware approach.  So, in order to have it included by default, we would provide a merged middleware, and document that as _the_ primary local signing middleware?
